lib/Sema/ConstraintGraph.cpp

```diff
@@ -1146,26 +1141,21 @@ bool ConstraintGraph::contractEdges() {
     auto rep1 = CS.getRepresentative(tyvar1);
     auto rep2 = CS.getRepresentative(tyvar2);
 
-    if (((rep1->getImpl().canBindToLValue() ==
-          rep2->getImpl().canBindToLValue()) ||
-         // Allow l-value contractions when binding parameter types.
-         isParamBindingConstraint)) {
-      if (CS.isDebugMode()) {
-        auto &log = llvm::errs();
-        if (CS.solverState)
-          log.indent(CS.solverState->depth * 2);
-
-        log << "Contracting constraint ";
-        constraint->print(log, &CS.getASTContext().SourceMgr);
-        log << "\n";
-      }
+    if (CS.isDebugMode()) {
+      auto &log = llvm::errs();
```

Comment:
Could you please adjust it to use `getDebugLogger()` instead of `llvm::errs()` here? It would also make sure that everything is indented properly without having to call `log.indent`.

Reply:
Good catch!

Reply:
Bah, it's on `Step`, so not quite so easy. I'm going to try to delete this code instead.